libyul/AsmAnalysis.cpp

```diff
@@ -352,77 +361,86 @@ size_t AsmAnalyzer::operator()(FunctionCall const& _funCall)
-	{
-		if (builtin->name == "selfdestruct")
-			m_errorReporter.warning(
-				1699_error,
-				nativeLocationOf(_funCall.functionName),
-				"\"selfdestruct\" has been deprecated. "
-				"Note that, starting from the Cancun hard fork, the underlying opcode no longer deletes the code and "
-				"data associated with an account and only transfers its Ether to the beneficiary, "
-				"unless executed in the same transaction in which the contract was created (see EIP-6780). "
-				"Any use in newly deployed contracts is strongly discouraged even if the new behavior is taken into account. "
-				"Future changes to the EVM might further reduce the functionality of the opcode."
-			);
-		else if (
-			m_evmVersion.supportsTransientStorage() &&
-			builtin->name == "tstore" &&
-			!m_errorReporter.hasError({2394})
-		)
-			m_errorReporter.warning(
-				2394_error,
-				nativeLocationOf(_funCall.functionName),
-				"Transient storage as defined by EIP-1153 can break the composability of smart contracts: "
-				"Since transient storage is cleared only at the end of the transaction and not at the end of the outermost call frame to the contract within a transaction, "
-				"your contract may unintentionally misbehave when invoked multiple times in a complex transaction. "
-				"To avoid this, be sure to clear all transient storage at the end of any call to your contract. "
-				"The use of transient storage for reentrancy guards that are cleared at the end of the call is safe."
-			);
-
-		numParameters = builtin->numParameters;
-		numReturns = builtin->numReturns;
-		if (!builtin->literalArguments.empty())
-			literalArguments = &builtin->literalArguments;
-
-		validateInstructions(_funCall);
-		m_sideEffects += builtin->sideEffects;
-	}
-	else if (m_currentScope->lookup(YulName{resolveFunctionName(_funCall.functionName, m_dialect)}, GenericVisitor{
-		[&](Scope::Variable const&)
-		{
-			m_errorReporter.typeError(
-				4202_error,
-				nativeLocationOf(_funCall.functionName),
-				"Attempt to call variable instead of function."
-			);
+	std::visit(
+		GenericVisitor {
+			[&](BuiltinName const& _builtinName)
+			{
+				BuiltinFunction const& builtin = m_dialect.builtin(_builtinName.handle);
+				if (builtin.name == "selfdestruct")
+					m_errorReporter.warning(
+						1699_error,
+						nativeLocationOf(_builtinName),
+						"\"selfdestruct\" has been deprecated. "
+						"Note that, starting from the Cancun hard fork, the underlying opcode no longer deletes the code and "
+						"data associated with an account and only transfers its Ether to the beneficiary, "
+						"unless executed in the same transaction in which the contract was created (see EIP-6780). "
+						"Any use in newly deployed contracts is strongly discouraged even if the new behavior is taken into account. "
+						"Future changes to the EVM might further reduce the functionality of the opcode."
+					);
+				else if (
+					m_evmVersion.supportsTransientStorage() &&
```

Comment:
Since `tstore` is only a builtin on EVMs that support transient storage, this part is redundant and could be just an assert inside the `if`.

Reply:
Ah yeah, right!!